e2e/cypress/integration/archived_channel/archive_channel_search_spec.js

```diff
@@ -0,0 +1,163 @@
+        cy.apiLogin(adminUser);
+
+        // # Post some text in the channel such as "Pineapple"
+        const messageText = `pineaple ${getRandomId()}`;
+
+        // # Archive the channel
+        createArchivedChannel({prefix: 'pineaple-'}, [messageText]).then(() => {
+            // # Unarchive the channel:
+            cy.uiUnarchiveChannel().then(() => {
+                // # remove search on archived channels
+                cy.apiUpdateConfig({
+                    TeamSettings: {
+                        ExperimentalViewArchivedChannels: false,
+                    },
+                });
+
+                cy.visit(`/${testTeam.name}/channels/off-topic`);
+                cy.contains('#channelHeaderTitle', 'Off-Topic');
+
+                // # Search for the post from step 1')
+                cy.get('#searchBox').focus().clear();
+                cy.get('#searchBox').should('be.visible').type(`${messageText}{enter}`);
+
+                // * Post is returned by search, since it's not archived anymore
+                cy.get('#searchContainer').should('be.visible');
+                cy.get('.search-item-snippet').first().contains(messageText);
+            });
+        });
+    });
+    it('MM-T1708 An archived channel can\'t be searched when "Allow users to view archived channels" is set to False in "Site Configuration > Users and Teams" in the System Console', () => {
+    // # First, as system admin, ensure that System Console > Users and Teams > Allow users to view archived channels is set to `false`.
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: false,
+            },
+        });
+
+        cy.apiLogin(testUser);
+
+        // # Open a channel other than Town Square
+        cy.visit(`/${testTeam.name}/channels/off-topic`);
+        cy.contains('#channelHeaderTitle', 'Off-Topic');
+
+        // next steps were performed previously
+        // # Create or locate a channel you're a member of
+        // # Post distinctive text in the channel such as "I like pineapples"
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.get('#channelHeaderTitle').should('be.visible');
+
+        // # Select Archive Channel from the header menu
+        cy.uiArchiveChannel();
+
+        // # Archive dialogue message reads"This will archive the channel from the team and make its contents inaccessible for all users" (Mobile dialogue makes no mention of the data will be accessible)
+        cy.get('#searchBox').focus().clear();
+        cy.get('#searchBox').focus().type(`${testArchivedMessage}{enter}`);
+
+        // * Post is not returned by search
+        cy.get('#searchContainer').should('be.visible');
+        cy.get('.no-results__wrapper').should('be.visible');
+    });
+
+    it('MM-T1709 Archive a channel while search results are displayed in RHS', () => {
+        const messageText = `search ${getRandomId()} pineaples`;
```

Comment:
Typo error on `pineapple`. Also, please fix other instance of it.
```suggestion
        const messageText = `search ${getRandomId()} pineapples`;
```